pkg/rke2/clusterrole_templates.go

```diff
@@ -0,0 +1,70 @@
+package rke2
```

Comment:
When @briandowns did the PSP work, as a non-essential review item I asked him to consider using structs instead of const strings. He did that for networkPolicy: https://github.com/rancher/rke2/blob/master/pkg/rke2/np.go#L28

Now having seen now seen both, I'm not sure which is right/better. The string consts are much more readable than the structs, but then instead of dealing with the "real" objects and clients in your code, you're dealing with yaml strings.

design tradeoffs amirght?

At any rate, either is fine for now and I would not want you to spend the time refactoring.

Reply:
Having done both now, the strings definitely yield better readability. Since this code is run only at start up, I'd rather have readability over one-off efficiency or proficiency.

Reply:
I prefer structs with helper functions but that does make for a lot more scaffolding code. Maybe unstructured types hit the sweet spot? https://godoc.org/github.com/kubernetes/apimachinery/pkg/apis/meta/v1/unstructured#Unstructured

Reply:
There are actually a bunch of helpers for creating RBAC entries with a builder pattern, as opposed to using raw strings or structs. We should probably be doing that instead. The end result less immediately visible than a nice chunk of YAML, but I think overall it's cleaner.

https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L105